src/js/editor/post.js

```diff
@@ -482,8 +528,8 @@ class PostEditor {
    * @public
    */
   insertSectionBefore(section, beforeSection) {
-    this.editor.post.sections.insertBefore(section, beforeSection);
-    this.editor.post.renderNode.markDirty();
+    section.parent.sections.insertBefore(section, beforeSection);
```

Comment:
This should use `beforeSection.parent` instead of `section.parent`. The `section` may already have a parent, but it would not be parent you want to insert into.